lib/Timer/src/timer.h

```diff
@@ -0,0 +1,32 @@
+#ifndef TIMER_H
+#define TIMER_H
+
+#include <Arduino.h>
+
+/**
+ * This class represents a parent option i.e an option that can hold n number of children options.
+ **/
+class Timer {
+
+    private:
+        unsigned int nextTimeout = 0;
```

Comment:
It's recommended in the C++ coding standards to rather use the constructor to initialize member fields. A default value of 0 for the field *nextTimeOut* can be set like this in the constructor: *Timer(usigned int nextTimeOut = 0)*. I also just read that it is conventional to put an underscore in front of private member fields, like this: _nextTimeOut.

Reply:
A second thing, if you are going to use *millis()* to set the value of this field, I would rather use the 4 byte data type *unsigned long* since millis() will overflow the unsigned int (max value is 2^16 - 1 = 65,535) in about 65 seconds. The unsigned long have a max value of 2^32 - 1, and *millis()* will become this big after 50 days ish of continuous runtime :P